src/sys/socket/ffi.rs

```diff
@@ -61,6 +61,22 @@ pub struct cmsghdr {
     pub cmsg_data: [type_of_cmsg_data; 0]
 }
 
+// See /usr/include/net/if_dl.h
+// sdl_data does not match if_dl.h on OS X, since the size of 12 is a minimum.
+// Will be unsafe
+// when sdl_nlen > 40.
+#[cfg(any(target_os = "freebsd", target_os = "macos"))]
+pub struct sockaddr_dl {
```

Comment:
Please read the `CONTRIBUTING.md` document. Structs should not be declared here, but instead in `libc` directly.

Reply:
this struct from [apple/xnu](https://github.com/apple/darwin-xnu/blob/master/bsd/net/if_dl.h#L95) , and not defined in `libc` .

so, what can i do?

any suggestion ?

Reply:
It's likely fine to add to `libc`. `libc` has been including types that aren't strictly in `libc` for a while now as long as they're included in system headers. You can make an issue about it in that repo first if you'd like to make sure, but it's likely fine to include. Also note that this will need to work on more than just these 2 platforms for this to get merged.

Reply:
okay, see https://github.com/rust-lang/libc/pull/866